### PR TITLE
fix: #WB-3350 issue with editing form in NCT timezone

### PR DESCRIPTION
**Pull request by Romu-C**
## Describe your changes

One boolean was used to display the loading when we open the modal to add new element. The boolean was used for both saving the form and displaying the modal. We now have two different boolean for that so the opening of the modal isn't impacting by the saving form in the back.

On the backend level, saving a form was not working for New Caledonia because of the date format used (the timezone shortcut NCT used for new caledonia is not handled by postgres).
The backend now saves the dates with an offset format that keeps the timezone information and is processable by postgres no matter the local timezone of the server.

## Checklist tests

Reproduced locally with faking New Caledonia timezone on local springboard.

The fix on the backend level required some refactoring of the date handling (now handling ZonedDateTime) that affect specifically the dateOpening and dateEnding of forms.
Unit test have been adapted and now all pass no matter the timezone of the server it is executed on.
However, a specific attention will probably be necessary from the QA team to verify that functional features linked to Opening and Ending dates are still working.

## Issue ticket number and link

https://edifice-community.atlassian.net/browse/WB-3350

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java [hunk 1]
```diff
@@ -713,28 +707,12 @@ private Future<String> getFormPublicKey(JsonObject form) {
...
-            }
-            catch (ParseException e2) {
-                e2.printStackTrace();
-                String errorMessage = "Failed to parse dates";
-                log.error("[Formulaire@DefaultFormService::getFormPublicKey] " + errorMessage);
-                promise.fail(e2.getMessage());
-                return promise.future();
-            }
+        catch (DateTimeParseException e) {
+            e.printStackTrace();
```
Comment (juniorode) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java:711]: Plutôt `log.error("gna gna gna", e);`

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java [hunk 2]
```diff
@@ -713,28 +707,12 @@ private Future<String> getFormPublicKey(JsonObject form) {
...
+            String errorMessage = "Failed to parse dates";
+            log.error("[Formulaire@DefaultFormService::getFormPublicKey] " + errorMessage);
+            promise.fail(e.getMessage());
```
Comment (juniorode) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java:714]: Tu peux faire `promise.fail(e)` directement.

**Review by juniorode — CHANGES_REQUESTED** [review]

File: common/src/main/java/fr/openent/form/core/models/Form.java [hunk 3]
```diff
@@ -262,5 +268,31 @@ public JsonObject toJson() {
...
+	    if (inputDateStr != null) {
+		    if (inputDateStr.endsWith("Z")) {
+			    try {
+				    formattedDate = Optional.of(ZonedDateTime.parse(inputDateStr));
+			    } catch (DateTimeParseException zonedDateParsingError) {
+				    log.error("[Form] Failed parsing zoned date = " + inputDateStr, zonedDateParsingError);
+			    }
+		    } else {
+			    try {
+				    formattedDate = Optional.of(LocalDateTime.parse(inputDateStr, DateTimeFormatter.ofPattern(DateFormats.YYYY_MM_DD_T_HH_MM_SS_SSS)).atZone(ZoneId.systemDefault()));
```
Comment (jcbe-ode) [inline common/src/main/java/fr/openent/form/core/models/Form.java:289]: Pas besoin de reconstruire le formatter à chaque appel car d'[après la doc](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#pattern)
> A formatter created from a pattern can be used as many times as necessary, it is immutable and is thread-safe.

=> Autant le construire 1 seule fois.

**Review by jcbe-ode — COMMENTED** [review]
A part un détail mineur, ok pour moi

**Pull request merged**